cluster/src/dunit/scala/io/snappydata/cluster/CassandraSnappyDUnitTest.scala

```diff
@@ -0,0 +1,223 @@
+/*
+ * Copyright (c) 2018 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.cluster
+
+import java.io.{BufferedWriter, ByteArrayInputStream, File, FileFilter, FileOutputStream, OutputStreamWriter, PrintWriter}
+import java.util
+
+import scala.sys.process._
+import scala.language.postfixOps
+
+import io.snappydata.test.dunit.DistributedTestBase
+import io.snappydata.test.dunit.DistributedTestBase.WaitCriterion
+import org.apache.commons.io.FileUtils
+import org.apache.commons.io.filefilter.{IOFileFilter, TrueFileFilter, WildcardFileFilter}
+
+import org.apache.spark.{Logging, TestPackageUtils}
+
+class CassandraSnappyDUnitTest(val s: String)
+    extends ClusterManagerTestBase(s)
+        with Logging {
+  // scalastyle:off println
+
+  private val snappyProductDir = System.getenv("SNAPPY_HOME")
+
+  val scriptPath = s"$snappyProductDir/../../../cluster/src/test/resources/scripts"
+
+  val currDir = System.getProperty("user.dir")
+
+  val userHome = System.getProperty("user.home")
+
+  var cassandraClusterLoc = ""
+
+  private val commandOutput = "command-output.txt"
+
+  def snappyShell: String = s"$snappyProductDir/bin/snappy-sql"
+
+  override def beforeClass(): Unit = {
+
+    logInfo(s"Starting snappy cluster in $snappyProductDir/work")
+
+    logInfo((snappyProductDir + "/sbin/snappy-start-all.sh").!!)
+
+    val start = System.currentTimeMillis
+    "curl -OL http://www-us.apache.org/dist/cassandra/2.1.21/apache-cassandra-2.1.21-bin.tar.gz".!!
```

Comment:
Should check if it already exists and download only if required. It's about 24 MB in size.

Reply:
Added change for searching folder if already present otherwise download.